src/Symfony/Component/HttpKernel/DependencyInjection/FragmentRendererPass.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\DependencyInjection;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+
+/**
+ * Adds services tagged kernel.fragment_renderer as HTTP content rendering strategies.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+class FragmentRendererPass implements CompilerPassInterface
+{
+    private $handlerService;
+    private $fragmentTag;
+
+    /**
+     * @param string $handlerService Service name of the fragment handler in the container
+     * @param string $fragmentTag    Tag name used for fragments
+     */
+    public function __construct($handlerService = 'fragment.handler', $fragmentTag = 'kernel.fragment_renderer')
+    {
+        $this->handlerService = $handlerService;
+        $this->fragmentTag = $fragmentTag;
+    }
+
+    public function process(ContainerBuilder $container)
+    {
+        if (false === $container->hasDefinition($this->handlerService)) {
+            return;
+        }
+
+        $definition = $container->getDefinition($this->handlerService);
```

Comment:
should we support the case of the handler service being an alias or no ? I would say no (as this assumes a method which is not part of the interface), but this would mean we should define our service as a private service and then define ``fragment.handler`` as an alias for it, using the private id in the compiler pass, to be sure that this compiler pass does not break if someone uses service decoration on ``fragment.handler``